### DISCUSSION TITLE
Decimation with custom vertex attributes

### DISCUSSION THREAD

**Discussion by Zeta48** [libigl/libigl#2119] [Ideas]
I am looking for a mesh simplification library to add LOD system to my engine.
Decimation in  libigl uses vertex and index buffers as input which is perfect. 

What I am missing is vertex attributes(eg: normals, texture coordinates). Please would it be possible to add vertex attributes support?
Decimation will become:
`
    igl::decimate(V, F, A, <target_nb_faces>, U, G, B);

`

A will stand for the input attribute matrix while B is output attributes.
A will be a [nxm] matrix.
B will be a [kxm] matrix.
With:
m : sum of number of components of all attributes. For eg equals 5 for a normal and texture coordinates.
n  : being the number of vertices in V.
k  : being the number of vertices in U.

Thanks for helping!

**Comment by alecjacobson** [comment]
doing this right is tricky (most modeling tools don't even do it right). The best code out there might be this decimator derived from libigl's https://github.com/songrun/SeamAwareDecimater this will correctly handle UV seams and data living at corners/wedges.

It would be awesome to eventually merge the SeamAwareDecimater into libigl and maintain it. Though I don't see that happening soon without a helpful PR.

meanwhile, if all of your data is per-vertex and you don't care about seams one "hot fix" for the libigl decimator is to concatenate all attributes:

A← [V UV N etc.]

and call decimate(A,F,…) instead of decimate(V,F,…)
